src/libraries/Native/Unix/System.Native/pal_networkchange.c

```diff
@@ -23,7 +23,9 @@
 #error System must have linux/rtnetlink.h or net/route.h.
 #endif
 
+#ifdef __clang__
```

Comment:
Why is this needed? It seems that gcc happily ignores clang pragmas (tried with gcc 9.3)

Reply:
On Clear Linux, it says:

```sh
/runtime/src/libraries/Native/Unix/System.Native/pal_networkchange.c:26: error: ignoring #pragma clang diagnostic [-Werror=unknown-pragmas]
   26 | #pragma clang diagnostic ignored "-Wcast-align" // NLMSG_* macros trigger this
      | 
```

```sh
> echo $CFLAGS

-g -O3 -feliminate-unused-debug-types -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=32 -Wformat -Wformat-security -m64 -fasynchronous-unwind-tables -Wp,-D_REENTRANT -ftree-loop-distribute-patterns -Wl,-z -Wl,now -Wl,-z -Wl,relro -fno-semantic-interposition -ffat-lto-objects -fno-trapping-math -Wl,-sort-common -Wl,--enable-new-dtags -mtune=skylake -Wa,-mbranches-within-32B-boundaries

> echo $CXXFLAGS

-g -O3 -feliminate-unused-debug-types -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=32 -Wformat -Wformat-security -m64 -fasynchronous-unwind-tables -Wp,-D_REENTRANT -ftree-loop-distribute-patterns -Wl,-z -Wl,now -Wl,-z -Wl,relro -fno-semantic-interposition -ffat-lto-objects -fno-trapping-math -Wl,-sort-common -Wl,--enable-new-dtags -mtune=skylake -Wa,-mbranches-within-32B-boundaries -fvisibility-inlines-hidden -Wl,--enable-new-dtags
```

This PR is only replacing `-D_FORTIFY_SOURCE=2 ` with empty string in `eng/native/configureplatform.cmake`. The rest of the flags are requiring these code changes.

Reply:
I am using latest clearlinux docker:

```sh
docker run -it clearlinux
```

Reply:
Ah, I have not used the -Wall when trying it. Makes sense then.

Reply:
Perhaps we just want to put -Wnoerror=unknown-pragmas in the master makefile, then?

Reply:
I agree with @bartonjs . I have zero sympathy for noisy C/C++ warnings that do not help you write better code. We are spending a ton of time of fixing, working around and reviewing the fixes for these. We should be rather using `-WnoXXX` more often.

Reply:
Regarding this particular change, the next line is:

```c
 #pragma clang diagnostic ignored "-Wcast-align" // NLMSG_* macros trigger this
```

and this one is saying it is only valid for clang. I think fixing the cast-alignment would be more valuable?
From the diff of this PR, we can see that (the much larger) CoreCLR code is written without needing to having such inline suppressions.

Reply:
> we can see that (the much larger) CoreCLR code is written without needing to have such inline suppressions.

That's because we have a lot more warnings disabled in CoreCLR

Reply:
>  I think fixing the cast-alignment would be more valuable?

That's hard to do when the code comes from system header.

Reply:
I have suppressed `unknown-pragmas` for libraries.